ivadomed/uncertainty.py

```diff
@@ -17,41 +17,40 @@ def run_uncertainty(ifolder):
         ifolder (str): Folder containing the Monte Carlo samples.
     """
     # list subj_acq prefixes
-    subj_acq_lst = [f.split('_pred')[0] for f in os.listdir(ifolder)
-                    if f.endswith('.nii.gz') and '_pred' in f]
+    subj_acq_lst = [f.name.split('_pred')[0] for f in Path(ifolder).iterdir()
+                    if f.name.endswith('.nii.gz') and '_pred' in f.name]
     # remove duplicates
     subj_acq_lst = list(set(subj_acq_lst))
     # keep only the images where unc has not been computed yet
-    subj_acq_lst = [f for f in subj_acq_lst if not os.path.isfile(
-        os.path.join(ifolder, f + '_unc-cv.nii.gz'))]
+    subj_acq_lst = [f for f in subj_acq_lst if not Path(ifolder, f + '_unc-cv.nii.gz').is_file()]
 
     # loop across subj_acq
     for subj_acq in tqdm(subj_acq_lst, desc="Uncertainty Computation"):
         # hard segmentation from MC samples
-        fname_pred = os.path.join(ifolder, subj_acq + '_pred.nii.gz')
+        fname_pred = Path(ifolder, subj_acq + '_pred.nii.gz')
         # fname for soft segmentation from MC simulations
-        fname_soft = os.path.join(ifolder, subj_acq + '_soft.nii.gz')
+        fname_soft = Path(ifolder, subj_acq + '_soft.nii.gz')
         # find Monte Carlo simulations
-        fname_pred_lst = [os.path.join(ifolder, f)
-                          for f in os.listdir(ifolder) if subj_acq + '_pred_' in f and
-                          ('_painted' not in f) and ('_color' not in f)]
+        fname_pred_lst = [str(f)
+                          for f in Path(ifolder).iterdir() if subj_acq + '_pred_' in f.name and
+                          ('_painted' not in f.name) and ('_color' not in f.name)]
```

Comment:
I would recommend expand this out more over multiple lines with comments as the logic is at least 3 folds and is a bit hard to follow.